R/dtm.R

```diff
@@ -24,29 +52,35 @@ function(X, Grid, m0, r = 2, weight = 1) {
   # without weight
   if (length(weight) == 1) {
     X <- as.matrix(X)
-    weightBound <- m0 * NROW(X)
+    weightBound <- m0 * NROW(X) 
+    # use fast nearest neighbor search algorithm to find distances to k nearest neighbors
     knnDistance <- FNN::knnx.dist(
 		data = X, query = as.matrix(Grid), k = ceiling(weightBound),
 		algorithm = c("kd_tree"))
+    # utilize embedded Dtm function to find distance to measure
     return (Dtm(knnDistance = knnDistance, weightBound = weightBound, r = r))
 
   # with weight
   } else {
+    # establish the weightbound and weight parameters to be used in final DtmWeight function
```

Comment:
These and the next set are great comments.  The code below is somewhat non-obvious and the comments make it very easy to read what is happening.